sphinx/config.py

```diff
@@ -284,6 +279,12 @@ def __init__(self, dirname, filename, overrides, tags):
         # these two must be preinitialized because extensions can add their
         # own config values
         self.setup = config.get('setup', None)
+
+        if 'extensions' in overrides:
+            if isinstance(overrides['extensions'], string_types):
+                config['extensions'] = overrides.pop('extensions').split(',')
+            else:
+                config['extensions'] = overrides.pop('extensions')
```

Comment:
Does this totally overwrite the extensions or adds the new values to the config somewhere else in the code ? We'd probably want the later.